### PR TITLE
017_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_invalid_ImageType

### PR DESCRIPTION
**Pull request by maria-mih**
017_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_invalid_ImageType
Testing coverage available at #643
All common functions used available at #703
@istoimenova Please review

### CODE REVIEW

**Comment by istoimenova** [comment]
@maria-mih, please update script according to clarification: "Clarification about expected resut code when invalid <audioPassThruIcon> is sent"
@maria-mih, please reference the script in TestSet.

**Comment by maria-mih** [comment]
Script needs to be updated and renamed due to clarification of expected result when invalid parameter received.

**Comment by maria-mih** [comment]
Script needs to be updated so that logic is changed so that all checks for invalid image parameter are included when all parameters of request tested and imagetype = STATIC.

**Comment by maria-mih** [comment]
@istoimenova Please review after script is reworked. Please note that it is now negative test when ImageType is invalidated and all parameters are sent in PerformAudioPassThru request.

File: test_scripts/API/PerformAudioPassThru/017_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_invalid_ImageType.lua [hunk 1]
```diff
@@ -0,0 +1,110 @@
+---------------------------------------------------------------------------------------------
+-- Requirements summary:
+-- [PerformAudioPassThru] SDL must transfer request to HMI in case of valid "audioPassThruIcon" param
+-- [GeneralResultCodes] INVALID_DATA wrong type
+-- [Mobile API] PerformAudioPassThru request/response
+-- [HMI_API] [MOBILE_API] The "audioPassThruIcon" param at "ImageFieldName" struct
+--
+-- Description:
+-- In case that the request comes with
+-- wrong type (empty) parameter(including parameters of the structures)
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/017_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_invalid_ImageType.lua]: @maria-mih, please remove empty from description.

File: test_scripts/API/PerformAudioPassThru/017_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_invalid_ImageType.lua [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+-- SDL must respond with resultCode "INVALID_DATA" and success:"false"
+--
+-- 1. Used preconditions
+-- 1.1. Request is sent with audioPassThruIcon image structure with invalid ImageType parameter
+-- and valid value parameter, all other params used in PerformAudioPassThru_request
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/017_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_invalid_ImageType.lua]: @maria-mih, will be better to specify which invalid types will be checked: empty and integer.

File: test_scripts/API/PerformAudioPassThru/017_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_invalid_ImageType.lua [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+local commonPostconditions = require('user_modules/shared_testcases/commonPreconditions')
+local testCasesForPerformAudioPassThru = require('user_modules/shared_testcases/testCasesForPerformAudioPassThru')
+local commonTestCases = require('user_modules/shared_testcases/commonTestCases')
+local testCasesForPolicyTable = require('user_modules/shared_testcases/testCasesForPolicyTable')
+
+--[[ Local variables ]]
+local params_invalid_data =
+{  
+  {param_value = "", comment = "ImageType_is_empty"}, -- audioPassThruIcon with empty ImageType parameter
+  {param_value = 123, comment = "ImageType_wrong_type"}, --audioPassThruIcon contains wrong type (not predefined in enum STATIC/DYNAMIC) in ImageType parameter
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/017_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_invalid_ImageType.lua]: @maria-mih, will be better to comment "ImageType_wrong_type" like "ImageType_integer_type".
Also update comment "(not predefined in enum STATIC/DYNAMIC)" - the comment should be for integer value.

**Comment by maria-mih** [comment]
@istoimenova Script is updated please check commit df278c70078f2b351050eb5843dacec0334b80bb

File: test_scripts/API/PerformAudioPassThru/017_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_invalid_ImageType.lua (hunk 1 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/017_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_invalid_ImageType.lua]: @maria-mih, accepted.

File: test_scripts/API/PerformAudioPassThru/017_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_invalid_ImageType.lua (hunk 2 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/017_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_invalid_ImageType.lua]: @maria-mih, accepted.

File: test_scripts/API/PerformAudioPassThru/017_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_invalid_ImageType.lua (hunk 3 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/017_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_invalid_ImageType.lua]: @maria-mih, accepted.

**Pull request merged**